test/index.test.ts

```diff
@@ -183,15 +236,450 @@ describe('HipThrusTS', () => {
           ? true
           : false = true;
         // just make sure the other two exist cause the above ensures they're properly typed
-        expectType<any>(happyFamily.parent);
-        expectType<any>(happyFamily.child);
+        let happyFamilyParentExistType: typeof happyFamily.parent extends any
+          ? true
+          : false;
+        let happyFamilyChildExistType: typeof happyFamily.child extends any
+          ? true
+          : false;
+        // @ts-expect-error
+        happyFamilyParentExistType = false;
+        // @ts-expect-error
+        happyFamilyChildExistType = false;
       });
       it('gets all members after attachData() is called', async () => {
         await happyFamily.attachData();
         expect(happyFamily.childAge).to.be.equal(33);
       });
     });
   });
+  describe('Hipthrusts functional only', () => {
+    describe('HTPipeAttachData', () => {
+      it('attaches properly typed data from left and right sync data attacher', () => {
+        const left = {
+          attachData(context: { a: string }) {
+            return { b: 4 };
+          },
+        };
+
+        const rightFullyCovered = {
+          attachData(context: { b: number }) {
+            return { c: 4 };
+          },
+        };
+
+        const pipedAtoBC = HTPipeAttachData(left, rightFullyCovered);
+
```

Comment:
did you test to make sure our approach actually WORKS (i.e. FAILS when it should)?  For example, if you ADD a key to CorrectParam, or if you REMOVE a key?
- [ ] test the test?

Reply:
i checked your theory in HTPipe attach data tests
in case we have const left = {
attachData(context: { a: string }) {
return { b: 4 };
},
};

    const rightFullyCovered = {
      attachData(context: { b: number }) {
        return { c: 4 };
      },
    };
and interface CorrectParam {
a: string;
}
type CorrectReturnValue = PromiseOrSync<{ b: number; c: number }>;
if we add a one more key to correct param this one will return false instead of true
type ParamAssignableFromCorrect = Parameters<
typeof pipedAtoBC.attachData
>[0] extends CorrectParam
? true
: false;
if we remove one param this one will return false instead of true
type ParamAssignableToCorrect = CorrectParam extends Parameters<
typeof pipedAtoBC.attachData
>[0]
? true
: false;
but not both